src/lambdas/fusionauth-site-origin-handler/test/index.test.js

```diff
@@ -0,0 +1,148 @@
+import assert from 'node:assert/strict';
+import { handler } from '../src/index.js';
+
+let event, result;
+
+// Mock a lambda event that we can pass to the handler.
+const mockEvent = (path) => {
+  return { 'Records': [{ 'cf': { 'request': { 'headers': {}, 'uri': path }}}]}
+}
+
+// Build the expected response for a 301 redirect.
+const makeRedirect = (path) => {
+  return { 'status': 301, 'statusDescription': 'Moved', 'headers': {'location': [{ 'key': "Location", 'value': path }]}};
+}
+
+// Build the expected response for a passthrough.
+const makePassThroughRequest = (path) => {
+  return { headers: {}, uri: path };
+};
+
+
+// All the tests follow a similar format. We pass the mock event to the
+// handler which contains the path we're testing. The handler's real
+// response is asserted to be equal to the expected response.
+
+describe('html extension', function() {
```

Comment:
At some point we could simplify this with a function call like the following, just to reduce some of the duplicated boilerplate code. Not critical.
```
test('/something.html', makeRedirect('/something'));
```